source/common/http/filter/fault_filter.cc

```diff
@@ -149,11 +149,11 @@ void FaultFilter::abortWithHTTPStatus() {
   callbacks_->requestInfo().setResponseFlag(Http::AccessLog::ResponseFlag::FaultInjected);
 }
 
-bool FaultFilter::matchesTargetCluster(const HeaderMap& headers) {
+bool FaultFilter::matchesTargetCluster(const HeaderMap&) {
```

Comment:
You could get rid of the headers arg completely. It's used only to obtain the target cluster from connection manager.